ios/RNSentry.m

```diff
@@ -270,47 +270,34 @@ - (void)setEventEnvironmentTag:(SentryEvent *)event
               });
 }
 
-RCT_EXPORT_METHOD(captureEnvelope:(NSDictionary * _Nonnull)envelopeDict
+RCT_EXPORT_METHOD(captureEnvelope:(NSArray * _Nonnull)bytes
+                  options: (NSDictionary * _Nonnull)options
                   resolve:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject)
 {
-    if ([NSJSONSerialization isValidJSONObject:envelopeDict]) {
-        SentrySdkInfo *sdkInfo = [[SentrySdkInfo alloc] initWithDict:envelopeDict[@"header"]];
-        SentryId *eventId = [[SentryId alloc] initWithUUIDString:envelopeDict[@"header"][@"event_id"]];
-        SentryTraceContext *traceContext = [[SentryTraceContext alloc] initWithDict:envelopeDict[@"header"][@"trace"]];
-        SentryEnvelopeHeader *envelopeHeader = [[SentryEnvelopeHeader alloc] initWithId:eventId sdkInfo:sdkInfo traceContext:traceContext];
-
-        NSError *error;
-        NSData *envelopeItemData = [NSJSONSerialization dataWithJSONObject:envelopeDict[@"payload"] options:0 error:&error];
-        if (nil != error) {
-            reject(@"SentryReactNative", @"Cannot serialize event", error);
-        }
-
-        NSString *itemType = envelopeDict[@"payload"][@"type"];
-        if (itemType == nil) {
-            // Default to event type.
-            itemType = @"event";
-        }
-
-        SentryEnvelopeItemHeader *envelopeItemHeader = [[SentryEnvelopeItemHeader alloc] initWithType:itemType length:envelopeItemData.length];
-        SentryEnvelopeItem *envelopeItem = [[SentryEnvelopeItem alloc] initWithHeader:envelopeItemHeader data:envelopeItemData];
+    NSMutableData *data = [[NSMutableData alloc] initWithCapacity: [bytes count]];
+    for(NSNumber *number in bytes) {
+        char byte = [number charValue];
+        [data appendBytes: &byte length: 1];
+    }
 
-        SentryEnvelope *envelope = [[SentryEnvelope alloc] initWithHeader:envelopeHeader singleItem:envelopeItem];
+    SentryEnvelope *envelope = [PrivateSentrySDKOnly envelopeWithData:data];
```

Comment:
Can we validate that the envelope contains the `traceContext` when is a transaction? just need to be sure that we are not losing data during the deserialization.

Reply:
![Screenshot 2022-09-15 at 10 55 23](https://user-images.githubusercontent.com/31292499/190360961-d94d6a5a-5467-41a8-acf1-3cbb69c2aadf.png)

Looks good.